tests/clients/test_alerts.py

```diff
@@ -356,3 +367,23 @@ def test_alerts_client_get_all_alert_details_sorts_results_descending_when_speci
         query.sort_key = sort_key
         results = list(alert_client.get_all_alert_details(query))
         assert results == [ALERT_F, ALERT_E, ALERT_D, ALERT_C, ALERT_B, ALERT_A]
+
+    def test_alerts_client_get_all_alert_details_returns_empty_generator_when_no_alerts_found(
+        self, mock_alerts_service_with_no_alerts, mock_alert_rules_service
+    ):
+        alert_client = AlertsClient(
+            mock_alerts_service_with_no_alerts, mock_alert_rules_service
+        )
+        query = AlertQuery()
```

Comment:
This is mostly just a clarifying question for my own understanding - but the `AlertQuery` constructor returns the `all()` method usually right? - And this is just returns no results because the testing env doesn't have any alerts?